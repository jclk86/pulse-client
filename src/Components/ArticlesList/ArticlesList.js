import React, { Component } from "react";
import { withRouter } from "react-router-dom";
import ArticleListItem from "../ArticleListItem/ArticleListItem";
import PropTypes from "prop-types";

class ArticlesList extends Component {
  render() {
    const { sortedArticles } = this.props;

    return (
      <div className="container_article_list" role="navigation">
        {sortedArticles.map(article => (
          <ArticleListItem article={article} key={article.id}></ArticleListItem>
        ))}
      </div>
    );
  }
}

ArticlesList.propTypes = {
  article: PropTypes.shape({
    id: PropTypes.number,
    article_category: PropTypes.string,
    image_url: PropTypes.string,
    title: PropTypes.string,
    content: PropTypes.string,
    date_created: PropTypes.string,
    num_of_votes: PropTypes.number,
    author: PropTypes.shape({
      id: PropTypes.number,
      username: PropTypes.string,
      fullname: PropTypes.string
    })
  })
};

export default withRouter(ArticlesList);
