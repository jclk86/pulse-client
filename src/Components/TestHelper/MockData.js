export const mockUser = {
  username: "Pop123",
  password: "Password123!",
  fullname: "Bob Smith",
  email: "BSmith@gmail.com",
  image_url:
    "https://images.pexels.com/photos/1622419/pexels-photo-1622419.jpeg?auto=compress&cs=tinysrgb&dpr=2&h=750&w=1260",
  profile: "test profile"
};

export const mockArticle = {
  title: "test article",
  content: "test content",
  article_category: "News",
  id: 1,
  date_created: "2019-10-14T21:49:20.570Z",
  author: {
    username: mockUser.username
  }
};

export const mockComment = {
  id: 1,
  content: "test comment content",
  date_created: "2019-10-14T21:49:20.570Z",
  article_id: 1,
  user: {
    id: 1,
    username: mockArticle.author.username
  }
};

export const mockUsers = [
  {
    username: "Pop123",
    password: "Password123!",
    fullname: "Bob Smith",
    email: "BSmith@gmail.com",
    image_url:
      "https://images.pexels.com/photos/1622419/pexels-photo-1622419.jpeg?auto=compress&cs=tinysrgb&dpr=2&h=750&w=1260",
    profile: "test profile"
  },
  {
    username: "testuser2",
    password: "Password123!",
    fullname: "Test User",
    email: "TUser@gmail.com",
    image_url:
      "https://images.pexels.com/photos/1622419/pexels-photo-1622419.jpeg?auto=compress&cs=tinysrgb&dpr=2&h=750&w=1260",
    profile: "test profile"
  }
];

export const mockArticles = [
  {
    title: "test article one",
    content: "test content",
    article_category: "News",
    id: 1,
    date_created: "2019-10-14T21:49:20.570Z",
    author: {
      username: mockUsers[0].username
    }
  },
  {
    title: "test article two",
    content: "test content",
    article_category: "News",
    id: 2,
    date_created: "2019-10-14T21:49:20.570Z",
    author: {
      username: mockUsers[1].username
    }
  }
];

export const categories = [
  { name: "Diary" },
  { name: "Guide" },
  { name: "Advice" },
  { name: "Interview" },
  { name: "Random" },
  { name: "News" }
];
