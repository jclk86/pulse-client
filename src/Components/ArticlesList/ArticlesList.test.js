import React from "react";
import { BrowserRouter } from "react-router-dom";
import { mount } from "enzyme";
import toJson from "enzyme-to-json";
import AuthApiService from "../../Services/auth-api-service";
import ArticleApiService from "../../Services/article-api-service";
import { mockUsers, mockArticles } from "../TestHelper/MockData";
import ArticlesList from "./ArticlesList";

describe(`ArticlesList Component`, () => {
  beforeAll(() => {
    for (let i = 0; i < mockUsers.length; i++) {
      AuthApiService.postUser(mockUsers[i]).then(user => {
        AuthApiService.postLogin({
          username: user.username,
          password: user.password
        }).then(() => ArticleApiService.postArticle(mockArticles[i]));
      });
    }
  });

  it("renders ArticlesList component without issues", () => {
    const wrapper = mount(
      <BrowserRouter>
        <ArticlesList sortedArticles={mockArticles}></ArticlesList>
      </BrowserRouter>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
