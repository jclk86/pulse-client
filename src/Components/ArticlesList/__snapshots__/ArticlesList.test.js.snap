// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticlesList Component renders ArticlesList component without issues 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <withRouter(ArticlesList)
      sortedArticles={
        Array [
          Object {
            "article_category": "News",
            "author": Object {
              "username": "Pop123",
            },
            "content": "test content",
            "date_created": "2019-10-14T21:49:20.570Z",
            "id": 1,
            "title": "test article one",
          },
          Object {
            "article_category": "News",
            "author": Object {
              "username": "testuser2",
            },
            "content": "test content",
            "date_created": "2019-10-14T21:49:20.570Z",
            "id": 2,
            "title": "test article two",
          },
        ]
      }
    >
      <ArticlesList
        history={
          Object {
            "action": "POP",
            "block": [Function],
            "createHref": [Function],
            "go": [Function],
            "goBack": [Function],
            "goForward": [Function],
            "length": 1,
            "listen": [Function],
            "location": Object {
              "hash": "",
              "pathname": "/",
              "search": "",
              "state": undefined,
            },
            "push": [Function],
            "replace": [Function],
          }
        }
        location={
          Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          }
        }
        match={
          Object {
            "isExact": true,
            "params": Object {},
            "path": "/",
            "url": "/",
          }
        }
        sortedArticles={
          Array [
            Object {
              "article_category": "News",
              "author": Object {
                "username": "Pop123",
              },
              "content": "test content",
              "date_created": "2019-10-14T21:49:20.570Z",
              "id": 1,
              "title": "test article one",
            },
            Object {
              "article_category": "News",
              "author": Object {
                "username": "testuser2",
              },
              "content": "test content",
              "date_created": "2019-10-14T21:49:20.570Z",
              "id": 2,
              "title": "test article two",
            },
          ]
        }
      >
        <div
          className="container_article_list"
        >
          <withRouter(ArticleListItem)
            article={
              Object {
                "article_category": "News",
                "author": Object {
                  "username": "Pop123",
                },
                "content": "test content",
                "date_created": "2019-10-14T21:49:20.570Z",
                "id": 1,
                "title": "test article one",
              }
            }
            key="1"
          >
            <ArticleListItem
              article={
                Object {
                  "article_category": "News",
                  "author": Object {
                    "username": "Pop123",
                  },
                  "content": "test content",
                  "date_created": "2019-10-14T21:49:20.570Z",
                  "id": 1,
                  "title": "test article one",
                }
              }
              history={
                Object {
                  "action": "POP",
                  "block": [Function],
                  "createHref": [Function],
                  "go": [Function],
                  "goBack": [Function],
                  "goForward": [Function],
                  "length": 1,
                  "listen": [Function],
                  "location": Object {
                    "hash": "",
                    "pathname": "/",
                    "search": "",
                    "state": undefined,
                  },
                  "push": [Function],
                  "replace": [Function],
                }
              }
              location={
                Object {
                  "hash": "",
                  "pathname": "/",
                  "search": "",
                  "state": undefined,
                }
              }
              match={
                Object {
                  "isExact": true,
                  "params": Object {},
                  "path": "/",
                  "url": "/",
                }
              }
            >
              <div
                className="container_article_list_item"
              >
                <div
                  className="container_article_preview"
                >
                  <div
                    className="container_vote_arrows"
                  >
                    <div
                      className="container_arrow_up"
                    >
                      <span
                        className="voting_request_error"
                      />
                      <div
                        className="arrow_up"
                        onClick={[Function]}
                      />
                    </div>
                    <div
                      className="container_vote_count"
                    >
                      <p
                        className="vote_count"
                      >
                        0
                      </p>
                    </div>
                    <div
                      className="container_arrow_down"
                    >
                      <div
                        className="arrow_down"
                        onClick={[Function]}
                      />
                    </div>
                  </div>
                  <div
                    className="container_article_text_preview"
                  >
                    <NavLink
                      className="article_list_item_title_link"
                      role="navigation"
                      to="/articles/1"
                    >
                      <Link
                        aria-current={null}
                        className="article_list_item_title_link"
                        role="navigation"
                        to={
                          Object {
                            "hash": "",
                            "pathname": "/articles/1",
                            "search": "",
                            "state": null,
                          }
                        }
                      >
                        <LinkAnchor
                          aria-current={null}
                          className="article_list_item_title_link"
                          href="/articles/1"
                          navigate={[Function]}
                          role="navigation"
                        >
                          <a
                            aria-current={null}
                            className="article_list_item_title_link"
                            href="/articles/1"
                            onClick={[Function]}
                            role="navigation"
                          >
                             
                            <h4
                              className="article_list_item_title"
                            >
                              test article one
                            </h4>
                             
                          </a>
                        </LinkAnchor>
                      </Link>
                    </NavLink>
                    <p>
                      October 14, 2019
                       
                    </p>
                    <p>
                      test content
                    </p>
                    <p
                      className="article_info"
                    >
                      <span>
                        0
                         comments
                      </span>
                      <span>
                        /
                        News
                        /
                      </span>
                      <span>
                        by
                      </span>
                       
                      Pop123
                    </p>
                  </div>
                </div>
              </div>
            </ArticleListItem>
          </withRouter(ArticleListItem)>
          <withRouter(ArticleListItem)
            article={
              Object {
                "article_category": "News",
                "author": Object {
                  "username": "testuser2",
                },
                "content": "test content",
                "date_created": "2019-10-14T21:49:20.570Z",
                "id": 2,
                "title": "test article two",
              }
            }
            key="2"
          >
            <ArticleListItem
              article={
                Object {
                  "article_category": "News",
                  "author": Object {
                    "username": "testuser2",
                  },
                  "content": "test content",
                  "date_created": "2019-10-14T21:49:20.570Z",
                  "id": 2,
                  "title": "test article two",
                }
              }
              history={
                Object {
                  "action": "POP",
                  "block": [Function],
                  "createHref": [Function],
                  "go": [Function],
                  "goBack": [Function],
                  "goForward": [Function],
                  "length": 1,
                  "listen": [Function],
                  "location": Object {
                    "hash": "",
                    "pathname": "/",
                    "search": "",
                    "state": undefined,
                  },
                  "push": [Function],
                  "replace": [Function],
                }
              }
              location={
                Object {
                  "hash": "",
                  "pathname": "/",
                  "search": "",
                  "state": undefined,
                }
              }
              match={
                Object {
                  "isExact": true,
                  "params": Object {},
                  "path": "/",
                  "url": "/",
                }
              }
            >
              <div
                className="container_article_list_item"
              >
                <div
                  className="container_article_preview"
                >
                  <div
                    className="container_vote_arrows"
                  >
                    <div
                      className="container_arrow_up"
                    >
                      <span
                        className="voting_request_error"
                      />
                      <div
                        className="arrow_up"
                        onClick={[Function]}
                      />
                    </div>
                    <div
                      className="container_vote_count"
                    >
                      <p
                        className="vote_count"
                      >
                        0
                      </p>
                    </div>
                    <div
                      className="container_arrow_down"
                    >
                      <div
                        className="arrow_down"
                        onClick={[Function]}
                      />
                    </div>
                  </div>
                  <div
                    className="container_article_text_preview"
                  >
                    <NavLink
                      className="article_list_item_title_link"
                      role="navigation"
                      to="/articles/2"
                    >
                      <Link
                        aria-current={null}
                        className="article_list_item_title_link"
                        role="navigation"
                        to={
                          Object {
                            "hash": "",
                            "pathname": "/articles/2",
                            "search": "",
                            "state": null,
                          }
                        }
                      >
                        <LinkAnchor
                          aria-current={null}
                          className="article_list_item_title_link"
                          href="/articles/2"
                          navigate={[Function]}
                          role="navigation"
                        >
                          <a
                            aria-current={null}
                            className="article_list_item_title_link"
                            href="/articles/2"
                            onClick={[Function]}
                            role="navigation"
                          >
                             
                            <h4
                              className="article_list_item_title"
                            >
                              test article two
                            </h4>
                             
                          </a>
                        </LinkAnchor>
                      </Link>
                    </NavLink>
                    <p>
                      October 14, 2019
                       
                    </p>
                    <p>
                      test content
                    </p>
                    <p
                      className="article_info"
                    >
                      <span>
                        0
                         comments
                      </span>
                      <span>
                        /
                        News
                        /
                      </span>
                      <span>
                        by
                      </span>
                       
                      testuser2
                    </p>
                  </div>
                </div>
              </div>
            </ArticleListItem>
          </withRouter(ArticleListItem)>
        </div>
      </ArticlesList>
    </withRouter(ArticlesList)>
  </Router>
</BrowserRouter>
`;
